app/src/main/java/com/example/mpvcalculatorrx/mvp/presenter/CalculatorPresenter.java

```diff
@@ -0,0 +1,98 @@
+package com.example.mpvcalculatorrx.mvp.presenter;
+
+import android.app.Activity;
+
+import com.example.mpvcalculatorrx.mvp.model.CalculatorModel;
+import com.example.mpvcalculatorrx.mvp.view.CalculatorView;
+import com.example.mpvcalculatorrx.util.bus.RxBus;
+import com.example.mpvcalculatorrx.util.bus.observers.CharacterBusObserver;
+import com.example.mpvcalculatorrx.util.bus.observers.EqualsButtonObserver;
+import com.example.mpvcalculatorrx.util.bus.observers.IntegerBusObserver;
+import com.example.mpvcalculatorrx.util.bus.observers.ResetButtonObserver;
+
+import java.util.ArrayList;
+
+public class CalculatorPresenter extends MvpPresenter<CalculatorModel, CalculatorView> {
+    public CalculatorPresenter(CalculatorModel model, CalculatorView view) {
+        super(model, view);
+    }
+
+    public void register() {
+        Activity activity = view.getActivity();
+
+        if (activity == null) {
```

Comment:
Invert the logic to avoid call return and brake the code flux.

Reply:
Done